File: src/main/java/net/clgd/ccemux/emulation/tror/BlitLinePacket.java
package net.clgd.ccemux.emulation.tror;

public class BlitLinePacket extends TRoRPacket<LineData> {
	public BlitLinePacket(String text, String fg, String bg) {
		data = new LineData(text, fg, bg);
	}
	
	@Override
	public String getPacketCode() {
		return "TY";
	}
}

File: src/main/kotlin/net/clgd/ccemux/emulation/tror/BlitLinePacket.kt
package net.clgd.ccemux.emulation.tror

class BlitLinePacket(val text: String, val fg: String, val bg: String) : TRoRPacket<LineData>(LineData(text, fg, bg)) {
	override val packetCode: String = "TY"

	init {
		if (text.length != fg.length || fg.length != bg.length) {
			throw IllegalArgumentException("All arguments must have the same length")
		}
	}
}
